dlog/logger.go

```diff
@@ -84,6 +84,19 @@ type OptimizedLogger interface {
 	UnformattedLogf(level LogLevel, format string, args ...interface{})
 }
 
+// LoggerWithMaxLevel can be implemented by loggers that define a maximum
+// level that will be logged, i.e. if a logger defines a max-level of
+// LogLevelInfo, then only LogLevelError, LogLevelWarn, and LogLevelInfo will
+// be logged, and LogLevelDebug and LogLevelTrace will be discarded.
+//
+// This interface can be used for examining what the loggers max level is
+// so that resource consuming string formatting can be avoided if its known that
+// the resulting message will be discarded anyway.
+type LoggerWithMaxLevel interface {
+	// MaxLevel return the maximum loglevel that will be logged
+	MaxLevel() LogLevel
+}
```

Comment:
My only suggestion is that I'd consider including this in the `OptimizedLogger` interface rather than creating a separate interface. I'm not sure I *would*, but I'd consider it more deeply; and then include in the doc comment the reasoning for landing whichever way I did.

Reply:
(in other words: please document your rationale)